### PR TITLE
redo all of teams/* to use UserVersion rather than UID

### PR DESCRIPTION
**Pull request by maxtaco**
- this is mainly right, but there still might be some weird cases around account resets.
- also fix CORE-5503 while we are at it.

### CODE REVIEW

File: go/libkb/errors.go [hunk 1]
```diff
@@ -294,21 +294,6 @@ func (u NoKeyError) Error() string {
 	return "No public key found"
 }
 
-type NoEldestKeyError struct {
```
Comment (maxtaco) [inline go/libkb/errors.go]: These weren't being used, so I nuked them.

File: go/teams/teams.go [hunk 2]
```diff
@@ -561,12 +556,14 @@ func (t *Team) rotateBoxes(ctx context.Context, memSet *memberSet) (*PerTeamShar
 
 	// rotate the team key for all current members
 	existing, err := t.Members()
+	t.G().Log.Debug("existing: %+v", existing)
```
Comment (maxtaco) [inline go/teams/teams.go]: we can get rid of these, i'm surprised then snuck in.

Comment (patrickxb) [inline go/teams/teams.go]: get rid of what?

Comment (maxtaco) [inline go/teams/teams.go]: these debugs. leftover from the other branch

File: go/libkb/user.go [hunk 3]
```diff
@@ -93,6 +93,12 @@ func NewUserFromLocalStorage(g *GlobalContext, o *jsonw.Wrapper) (*User, error)
 func (u *User) GetNormalizedName() NormalizedUsername { return NewNormalizedUsername(u.name) }
 func (u *User) GetName() string                       { return u.name }
 func (u *User) GetUID() keybase1.UID                  { return u.id }
+func (u *User) GetUserVersion() keybase1.UserVersion {
```
Comment (patrickxb) [inline go/libkb/user.go]: `ToUserVersion()` is in this file.  should delete it.

File: go/teams/keys.go [hunk 4]
```diff
@@ -186,7 +186,7 @@ func (t *TeamKeyManager) recipientBoxes(secret []byte, nonce *nonce24, senderKey
 			return nil, err
 		}
 
-		boxes[uid] = base64.StdEncoding.EncodeToString(encodedArray)
+		boxes[uv.Uid] = base64.StdEncoding.EncodeToString(encodedArray)
```
Comment (patrickxb) [inline go/teams/keys.go]: change the map key to keybase1.UserVersion?

**Comment by keybase-ci-visdiff** [comment]
The commits 34372bc78cbd...34372a6d1b4f introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498231696841-34372bc78cbd-34372a6d1b4f-linux/Tracker-New_user_Scroll_1-34372bc78cbd.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498231696841-34372bc78cbd-34372a6d1b4f-linux/Tracker-New_user_Scroll_1-34372a6d1b4f.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 34372bc78cbd...62512206ef9c introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498232113634-34372bc78cbd-62512206ef9c-linux/Tracker-New_user_Scroll_1-34372bc78cbd.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498232113634-34372bc78cbd-62512206ef9c-linux/Tracker-New_user_Scroll_1-62512206ef9c.png)  
   Tracker-New_user_Scroll_1 comparison
 * Changed: **Markdown-emoji** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498232113634-34372bc78cbd-62512206ef9c-linux/Markdown-emoji-34372bc78cbd.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498232113634-34372bc78cbd-62512206ef9c-linux/Markdown-emoji-62512206ef9c.png)  
   Markdown-emoji comparison
</summary>

File: go/libkb/user.go (hunk 3 above)
Comment (maxtaco) [inline go/libkb/user.go]: ah cool thx

File: go/teams/keys.go (hunk 4 above)
Comment (maxtaco) [inline go/teams/keys.go]: Well right after this, we send up to the server with a map key via regular `UID`s. So we'd just have to down-convert it.

Comment (patrickxb) [inline go/teams/keys.go]: ok

**Comment by keybase-ci-visdiff** [comment]
The commits 34372bc78cbd...3e1a8625f8e3 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498233579500-34372bc78cbd-3e1a8625f8e3-linux/Tracker-New_user_Scroll_1-34372bc78cbd.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498233579500-34372bc78cbd-3e1a8625f8e3-linux/Tracker-New_user_Scroll_1-3e1a8625f8e3.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Pull request merged**